src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java

```diff
@@ -139,10 +141,75 @@ public void testParticipantCannotInsertOrUpdateOrDeleteQuestion() {
 
     lenient().when(modoRepo.findByToken("t1")).thenReturn(Optional.empty());
     lenient().when(participantRepository.findByToken("t1")).thenReturn(Optional.of(aloy));
-    //lenient().when(repo.delete(Mockito.any())).thenReturn(new ServerMessage("success"));
 
     assertThrows(WrongCredentialsException.class, () -> cc.insert(1, 123, "t1", c2));
     assertThrows(WrongCredentialsException.class, () -> cc.updateQuestion(1, 123, 1, "t1", c2));
     assertThrows(WrongCredentialsException.class, () -> cc.deleteQuestion(1, 123, 1, "t1"));
   }
+
+  @Test
+  public void testModeratorCanInsertOrUpdateQuestionOnlyWithRightAccess()
```

Comment:
Unit tests should test a single thing at a time (especially if they're independent). Could you please separate this method in two ?

Reply:
it's done :)